csrc/python_frontend/factory.h

```diff
@@ -0,0 +1,47 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+namespace nvfuser::python_frontend {
+
+// Flatbuffer enum are represented as an unscoped enumeration, so we can map
+// them to an Integer type. This Factory class contains a vector that maps from
+// an enum integer to its corresponding parser function.
+//
+// All parser functions have the same signature. We use std::bind to support
+// functions that require extra arguments.
+
+template <typename SerdeBuffer, typename BaseType>
```

Comment:
nitpick, do we need a template here?! Looks like there's only one instantiation.

Reply:
I was planning to use this class for the CPP Fusion / Kernel IR.